opacus/privacy_engine/factory.py

```diff
@@ -0,0 +1,33 @@
+from opacus import PrivacyEngine
+from opacus.privacy_engine.catalog import (
+    PrivacyEngine,
+    PrivacyEngineNonPoisson,
+    PrivacyEnginePerLayer,
+    PrivacyEnginePerLayerNonPoisson,
+)
+
+
+class PrivacyEngineFactory:
+    @classmethod
+    def get(
```

Comment:
nit: rename `get` to `create`?

Reply:
>If I understand correctly the idea, the class is redundant here. It's just a function.
Also Factory.get to my taste looks a little bit like a Java developer invaded the building.

Guilty as charged. You're right, this looks like a hardcore OOP pattern, we don't really need in here

> This get() looks like a constructor part absent in the implementation classes. If so, why not doing something like:

I'm not sure I understand what you mean by that.

Regardless, the subclass router option you've proposed look good to me. I'd be fine with either `build_privacy_engine()` function (replacing current factory) or subclass router in PrivacyEngine constructor.
I'm not sure I see a strong arguments for either - both looks ok?

cc @karthikprasad @alexandresablayrolles

Reply:
+1 to @romovpa's call out. I vote for a function to replace the factory and its method. A nit on the naming: `get_privacy_engine()` instead of `build_privacy_engine()` :)

> I'm not sure I see a strong arguments for either - both looks ok?

If not going ahead with factory pattern, I have a stronger argument in favour of function. The class here is neither acting on instance specific data with associated methods, nor is it a namespace; class doesn't add any value.

Reply:
Yup, that's a good argument - I'll switch to function.